src/transformers/models/whisper/processing_whisper.py

```diff
@@ -41,12 +41,46 @@ def __init__(self, feature_extractor, tokenizer):
         self.current_processor = self.feature_extractor
         self._in_target_context_manager = False
 
-    def __call__(self, *args, **kwargs):
+    def get_decoder_prompt_ids(self, task=None, language=None, no_timestamps=True, **kwargs):
+        forced_decoder_tokens = "<|startoftranscript|>"
+        if language is not None:
+            if language not in self.tokenizer.additional_special_tokens:
+                raise ValueError(
+                    f"{language} is not supported. The language should be one of the following: '<|en|>',"
+                    " '<|zh|>', '<|de|>', '<|es|>', '<|ru|>', '<|ko|>', '<|fr|>', '<|ja|>', '<|pt|>', '<|tr|>',"
+                    " '<|pl|>', '<|ca|>', '<|nl|>', '<|ar|>', '<|sv|>', '<|it|>', '<|id|>', '<|hi|>', '<|fi|>',"
+                    " '<|vi|>', '<|iw|>', '<|uk|>', '<|el|>', '<|ms|>', '<|cs|>', '<|ro|>', '<|da|>', '<|hu|>',"
+                    " '<|ta|>', '<|no|>', '<|th|>', '<|ur|>', '<|hr|>', '<|bg|>', '<|lt|>', '<|la|>', '<|mi|>',"
+                    " '<|ml|>', '<|cy|>', '<|sk|>', '<|te|>', '<|fa|>', '<|lv|>', '<|bn|>', '<|sr|>', '<|az|>',"
+                    " '<|sl|>', '<|kn|>', '<|et|>', '<|mk|>', '<|br|>', '<|eu|>', '<|is|>', '<|hy|>', '<|ne|>',"
+                    " '<|mn|>', '<|bs|>', '<|kk|>', '<|sq|>', '<|sw|>', '<|gl|>', '<|mr|>', '<|pa|>', '<|si|>',"
+                    " '<|km|>', '<|sn|>', '<|yo|>', '<|so|>', '<|af|>', '<|oc|>', '<|ka|>', '<|be|>', '<|tg|>',"
+                    " '<|sd|>', '<|gu|>', '<|am|>', '<|yi|>', '<|lo|>', '<|uz|>', '<|fo|>', '<|ht|>', '<|ps|>',"
+                    " '<|tk|>', '<|nn|>', '<|mt|>', '<|sa|>', '<|lb|>', '<|my|>', '<|bo|>', '<|tl|>', '<|mg|>',"
+                    " '<|as|>', '<|tt|>', '<|haw|>', '<|ln|>', '<|ha|>', '<|ba|>', '<|jw|>', '<|su|>'"
+                )
+            forced_decoder_tokens += f"<|{language}|>"
+
+        if task is not None:
+            if task not in self.tokenizer.additional_special_tokens:
+                raise ValueError(f"{task} is not supported. The language should be in : \{'transcribe', 'translate'}")
+            forced_decoder_tokens += f"<|{task}|>"
+
+        forced_decoder_tokens += "<|notimestamps|>" if no_timestamps else None
+        forced_decoder_ids = self.tokenizer.encode(forced_decoder_tokens, **kwargs)
```

Comment:
be careful here with how white spaces are handled think the `convert_token_to_ids` function is better

Reply:
`convert_tokens_to_ids` didn't work and only outputs the first token, I might be missing something